src/PostgREST/SchemaCache.hs

```diff
@@ -1203,6 +1206,34 @@ timezones = SQL.Statement sql HE.noParams decodeTimezones
     decodeTimezones :: HD.Result TimezoneNames
     decodeTimezones = S.fromList . map encodeUtf8 <$> HD.rowList (column HD.text)
 
+funcSettings :: PgVersion -> Bool -> SQL.Statement () FuncSettings
+funcSettings pgVer = SQL.Statement sql HE.noParams rows
+  where
+    sql = [q|
+      WITH
+      func_setting AS (
+        SELECT p.proname, unnest(p.proconfig) AS setting
+        FROM pg_proc p
+      ),
+      kv_settings AS (
+        SELECT
+          proname,
+          substr(setting, 1, strpos(setting, '=') - 1) as key,
+          lower(substr(setting, strpos(setting, '=') + 1)) as value
+        FROM func_setting
+      )
+      SELECT
+        proname, kv.key AS key, kv.value AS value
+        FROM kv_settings kv
+      JOIN pg_settings ps ON ps.name = kv.key |] <>
+      (if pgVer >= pgVersion150
+        then "and (ps.context = 'user' or has_parameter_privilege(current_user::regrole::oid, ps.name, 'set'));"
+        else "and ps.context = 'user';")
```

Comment:
For putting FuncSettings inside `Routine` you would need to move this query to `funcsSqlQuery`.